File: src/main/java/com/jss/osiris/modules/quotation/service/BodaccPublicationTypeService.java
package com.jss.osiris.modules.quotation.service;

import java.util.List;

import com.jss.osiris.modules.quotation.model.BodaccPublicationType;

public interface BodaccPublicationTypeService {
    public List<BodaccPublicationType> getBodaccPublicationTypes();

    public BodaccPublicationType getBodaccPublicationType(Integer id);

    public BodaccPublicationType addOrUpdateBodaccPublicationType(BodaccPublicationType bodaccPublicationType);
}

File: client/src/app/modules/quotation/services/bodacc-publication-type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { BodaccPublicationType } from '../../quotation/model/BodaccPublicationType';

@Injectable({
  providedIn: 'root'
})
export class BodaccPublicationTypeService extends AppRestService<BodaccPublicationType>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getBodaccPublicationTypes() {
    return this.getListCached(new HttpParams(), "bodacc-publication-types");
  }

  addOrUpdateBodaccPublicationType(bodaccPublicationType: BodaccPublicationType) {
    this.clearListCache(new HttpParams(), "bodacc-publication-types");
    return this.addOrUpdate(new HttpParams(), "bodacc-publication-type", bodaccPublicationType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
